### PR TITLE
*: DM server graceful exit in integration test

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
fix https://github.com/pingcap/dm/issues/49

### What is changed and how it works?
* add a channel to control `DM-worker`/`DM-master` server close
* wait all test main processes exit in each integration test case before next one
* ~fix some race condition~ fix in https://github.com/pingcap/dm/pull/53

Aflter change the coverage rate of DM increases from 52% to 57%

### Check List 

Tests 

 - Integration test
   run CI multiple times and observe coverage rate

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

**Comment by GregoryIan** [comment]
good catch！

**Comment by GregoryIan** [comment]
LGTM

File: cmd/dm-master/main.go [hunk 1]
```diff
@@ -71,3 +77,8 @@ func main() {
 
 	log.Info("dm-master exit")
 }
+
+func main() {
```
Comment (csuzhangxc) [inline cmd/dm-master/main.go]: not used anymore?

Comment (amyangfei) [reply cmd/dm-master/main.go]: main function is still the execution entrance of `dm-master`, `dm-worker`

File: cmd/dm-master/main_test.go [hunk 2]
```diff
@@ -34,19 +41,20 @@ func TestRunMain(t *testing.T) {
...
 
 	signalCh := make(chan os.Signal, 1)
 	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM, syscall.SIGHUP)
 	select {
-	case <-signalCh:
-		return
 	case <-waitCh:
-		return
+	case <-signalCh:
+		close(exitCh)
```
Comment (csuzhangxc) [inline cmd/dm-master/main_test.go]: our purpose is to control
1. exit the process when received signals
2. wait for the process exit
?

seems we call `signal.Notify` both in `main.go` and `main_test.go`. Is it needed?

File: tests/_utils/wait_process_exit [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+while true
+do
+    pgrep $process > /dev/null 2>&1
+    ret=$?
+    if [ "$ret" != "0" ]; then
+        echo "process $process already exit"
+        break
+    fi
+    sleep 0.1
+    echo "wait process $process exit..."
```
Comment (csuzhangxc) [inline tests/_utils/wait_process_exit]: `echo` every `0.1` second, too much log?

Comment (amyangfei) [reply tests/_utils/wait_process_exit]: changed to `0.2s`, not a lot of logs, processes often exit within 1s in these integration test

File: cmd/dm-master/main_test.go (hunk 2 above)
Comment (amyangfei) [reply cmd/dm-master/main_test.go]: well the test starts `main` in a single goroutine, so it catches singal in `main_test.go` too, which is not necessary. I simplify the code by call `main()` directly

**Pull request merged**